explorer/ast/declaration.h

```diff
@@ -90,6 +90,26 @@ class Declaration : public AstNode {
     return constant_value_;
   }
 
+  // Returns whether this node has been declared.
+  auto is_declared() const -> bool { return is_declared_; }
+
+  // Returns whether this node has been fully type-checked.
+  auto is_type_checked() const -> bool { return is_type_checked_; }
```

Comment:
Function definitions get a static type when the declaration is type-checked, which is before their body is type-checked, so that doesn't seem to work.